listener.go

```diff
@@ -16,17 +16,18 @@ var quicListenAddr = quic.ListenAddr
 
 // A listener listens for QUIC connections.
 type listener struct {
-	quicListener quic.Listener
-	transport    tpt.Transport
-
+	quicListener   quic.Listener
+	t              *transport
 	privKey        ic.PrivKey
 	localPeer      peer.ID
 	localMultiaddr ma.Multiaddr
+	udpAddr        *net.UDPAddr
+	network        string
```

Comment:
I'd mildly prefer to just re-parse the multiaddr on close (storing duplicate information bugs me).